### PR TITLE
fixes for some drupal/network tests

### PR DESCRIPTION
**Pull request by meek0**
Some tests for counts (notably variable counts) do not work.
Variable classifications are not shown.
Also did some method name refractors.

### CODE REVIEW

File: src/test/java/org/obiba/mica/uitest/drupal/network/NetworkAffiliatedMembers.java [hunk 1]
```diff
@@ -12,45 +13,46 @@
...
-    assertThatWeAreOnFirstPage();
+
+    // assert that we're on the first page
+    browser()
+        .element(By.xpath("//div[@id='person-table_paginate']/ul[contains(@class, 'pagination')]/li[contains(@class, 'paginate_button active')]/a")).hasText("1");
+
+    String paginationNumberButtonsXpath = "//div[@id='person-table_paginate']/ul[contains(@class, 'pagination')]/li[not(@id='person-table_next' or @id='person-table_previous')]/a";
+    int numberOfPages = browser().wrappedDriver().findElements(By.xpath(paginationNumberButtonsXpath)).size();
+
+    if (numberOfPages > 1) {
```
Comment (firone) [inline src/test/java/org/obiba/mica/uitest/drupal/network/NetworkAffiliatedMembers.java]: we are in a test, do we really need check ? isn't it a fail if the number is not correct ?

File: src/test/java/org/obiba/mica/uitest/drupal/network/NetworkAffiliatedMembers.java [hunk 2]
```diff
@@ -63,22 +65,39 @@ public void validate_modal_close_in_network_members_popup() throws Exception {
 
   private void networkOpenMembersPopup() {
     browser().navigate().to("http://localhost/drupal/mica/networks");
-    browser().element(By.xpath("(//h4)[5]")).hasText("QSC - Quebec Study Catalogue");
 
-    browser().element(By.xpath("(//h4)[5]/a")).click();
+    int randomXpathIndexForSize = getRandomXpathIndexForSize(
+        browser().wrappedDriver().findElements(By.xpath("//h4/a")).size());
+
+    String titleXpath = "(//h4/a)[" + randomXpathIndexForSize + "]";
```
Comment (firone) [inline src/test/java/org/obiba/mica/uitest/drupal/network/NetworkAffiliatedMembers.java]: i understand nothing in this test ...

which title xpath ?
why a random data ? where in a test, we need to know what we're testing ...

the purpose of a test is to check that expected data is here.
Like : assertThat(findContacts(), contains("Toto", "Titi"))
Not : assertThat(findContacts(), contains(contactRepository.findAll()))

A test should have as little dynamic content as possible.

In this case with a random data ... the test can fail sometime ... then will work ... a test must always fails or always works. rerun test 3 times need to show 3 times the same result.

File: src/test/java/org/obiba/mica/uitest/drupal/network/NetworkAffiliatedMembers.java [hunk 3]
```diff
@@ -63,22 +65,39 @@ public void validate_modal_close_in_network_members_popup() throws Exception {
...
+  private void validateSearchInNetworkMembersPopup(String column, int columnIndex) throws Exception {
+    networkOpenMembersPopup();
+    browser().pause(500);
 
-  private void assertThatWeAreOnFirstPage() {
-    browser().element(By.xpath("(//div[@id='person-table_wrapper']//td)[1]")).hasText("Canadian Longitudinal Study on Aging");
-  }
+    int size = browser().wrappedDriver().findElements(By.xpath(column)).size();
+
+    if (size > 0) {
```
Comment (firone) [inline src/test/java/org/obiba/mica/uitest/drupal/network/NetworkAffiliatedMembers.java]: same here, if you size > 0 there is a lot of thing to do but if size == 0 it's correct ? the test says yes.
impossible to be correct with 1 and 0

**Comment by meek0** [comment]
Removed most of the randomization, however, I kept those where the test should work for any random item (in the case of the table filters for example).

**Comment by meek0** [comment]
Added fix for drupal study tests.
Some test failures are due to search results (counts), will need to investigate those.

**Pull request merged**